kratos/python_scripts/model_parameters_factory.py

```diff
@@ -0,0 +1,88 @@
+    '''
+
+    def __init__(self, model):
+        self.model = model
+
+    def ConstructListOfItems( self, items_list):
+        constructed_items = []
+
+        for item in items_list:
+
+            # Check the input settings
+            # Note that we support both 'parameters' and 'Parameters' to keep processes backward compatibility
+            if item.Has("parameters"):
+                parameters = item["parameters"]
+            elif item.Has("Parameters"):
+                #TODO: Eventually throw a deprecation warning in here
+                parameters = item["Parameters"]
+            else:
+                error_msg = f"Provided item '{item}' is incomplete.\n"
+                error_msg += "'parameters' field must be defined for each item."
+                raise NameError(error_msg)
+
+            # Check for the 'name' entry
+            # Note that this is a registry-based factory so 'name' must be always provided
+            if item.Has("name"):
+                """Location of the item in Kratos; e.g.:
+                - Registered operation:                     operations.KratosMultiphysics.Operation
+                - Registered application process:           processes.KratosMultiphysics.FluidDynamicsApplication.FluidProcess
+                - Full path to Kratos modeler:              KratosMultiphysics.modelers.delete_model_parts_modeler.DeleteModelPartsModeler
+                - Full path to custom operation:            user_operation.UserOperation (this has to be located in the simulation directory)
+                """
+
+                # Check if current item is registered
+                # If not found, search for the corresponding Python module
+                registry_entry = item["name"].GetString()
+                if KM.Registry.HasItem(registry_entry):
+                    # Get already stored prototype
+                    if KM.Registry.HasItem(f"{registry_entry}.Prototype"):
+                        prototype = KM.Registry[f"{registry_entry}.Prototype"]
+                        instance = prototype.Create(self.model, parameters)
+                    # Get the create function
+                    elif KM.Registry.HasItem(f"{registry_entry}.CreateFunction"):
+                        create = KM.Registry[f"{registry_entry}.CreateFunction"]
+                        instance = create(self.model, parameters)
+                    # Get prototype from stored Python module
+                    elif KM.Registry.HasItem(f"{registry_entry}.ModuleName"):
+                        class_name = registry_entry.split(".")[-1]
+                        module_name = KM.Registry[f"{registry_entry}.ModuleName"]
+                        module = import_module(module_name)
+                        # Check if the standard class name is available
+                        # Note that in here we assume that the registry last key matches the class name
+                        if hasattr(module, class_name):
```

Comment:
Are you sure that if shouldn't be first if has ClassName?

I mean if we want to register two classes in the same module in this way it won't work...

Reply:
mmm... I think I'm not getting your point at all.

For the registry entry
```
"Processes.KratosMultiphysics.FluidDynamicsApplication.ApplyInletProcess" : {
    "ModuleName" : "KratosMultiphysics.FluidDynamicsApplication.apply_inlet_process"
}
```
`module_name` would be `"KratosMultiphysics.FluidDynamicsApplication.apply_inlet_process"`and module is the result of importing that. Then we retrieve the `class_name` from the registry entry last word, which in this case is `ApplyInletProcess`. Alternatively, we give the option of providing the `ClassName` together with the `ModuleName`.

Having said so, I think that this should work provided that there is no more than one class with the same name, which is something that according to my understanding we've always assumed in Kratos.

Reply:
Adding to my comment that I've just checked (again) that this mechanism works with that very same `ApplyInletProcess` and a CFD simulation.

Reply:
OK, I see. BTW, this can be revised easily afterward.